endoreg_db/models/media/video/video_file_anonymize.py

```
@@ -0,0 +1,354 @@
+def _censor_outside_frames(video: "VideoFile", outside_label_name: str = "outside", censor_color: Tuple[int, int, int] = (0, 0, 0)) -> bool:
+    """
+    Overwrites frame files marked as 'outside' with a censored version (e.g., black).
+    This modifies the original raw frames directly. Use with caution.
+    """
+    logger.warning("Starting direct censoring of 'outside' frames for video %s. This modifies raw frame files.", video.uuid)
+    state = video.get_or_create_state()
+    if not state.frames_extracted:
+        logger.error("Frames not extracted for video %s. Cannot censor.", video.uuid)
+        return False
+
+    outside_frames = _get_outside_frames(video, outside_label_name)
```

Ruff (0.8.2)

Undefined name `_get_outside_frames`

(F821)